scripts/scons_helpers/build-3rdparty.py

```diff
@@ -1471,10 +1490,26 @@ def die(text, *args):
         'gengetopt-{ctx.pkg_ver}.tar.gz',
         'gengetopt-{ctx.pkg_ver}')
     changedir(ctx, 'src/gengetopt-{ctx.pkg_ver}')
-    execute(ctx, './configure', clear_env=True)
+    compiler_launcher = ctx.env.get('COMPILER_LAUNCHER', None)
+    execute(ctx, './configure CC="{compiler_launcher}{CC_compiler}" CXX="{compiler_launcher}{CXX_compiler}"'.format(
+    compiler_launcher='%s' % (compiler_launcher + ' ') if compiler_launcher else '',
+    CC_compiler=CC_compiler,
+    CXX_compiler=CXX_compiler),
```

Comment:
Quoting is not quite correct because is provided by user and may have any symbols (e.g. `"`).

We already have `format_vars()` that handles both quoting and COMPILER_LAUNCHER, I suggest to reuse it.

Currently it formats vars from ctx.env, but we could add an optional argument `env`, unset by default. When it's set, it would use it instead of ctx.env.

Then we could use it like this:

```
format_vars(..., env={'CC': CC_compiler, 'CXX': ..., 
   'COMPILER_LAUNCHER': ctx.env['COMPILER_LAUNCHER']})
```

Reply:
Done, reused `format_vars` by adding a new parameter `env` which is set to `None` by default and is used instead of `ctx.env` when used.